test/dynamo/test_logging.py

```diff
@@ -746,6 +746,41 @@ def fn(a):
                     empty_line_normalizer(stderr.decode("utf-8")),
                 )
 
+    @make_settings_test("torch._dynamo.eval_frame")
+    def test_log_traced_frames(self, records):
+        # Test program
+        @torch.compile()
+        def foo():
+            x = torch.ones([10])
+
+            def bar():
+                y = x + x
+                torch._dynamo.graph_break()
+                z = y * x
+                return z
+
+            return bar(), bar
+
+        foo()
+
+        # `_log_traced_frames` is registered as an atexit callback, so we invoke
+        # it explicitly for testing.
+        torch._dynamo.eval_frame._log_traced_frames()
+
+        # Prepare expected strings.
+        record = self.getRecord(records, "TorchDynamo attempted to trace")
+        lines = record.getMessage().split("\n")
+        filename = os.path.basename(__file__)
+
+        # Check
+        self.assertEqual(len(lines), 4)
```

Comment:
You can try using assertExpectedInline instead of manually checking

Reply:
Thanks. I looked into this, seems like it's mainly helpful for automatically filling in the expected output with `EXPECTTEST_ACCEPT=1`? In this case we can't hard-code the output because the trace includes absolute filepath, that's also why I used regex below.

Reply:
There is a way to get ride of the absolute filepath, `munge_exc` I believe?

Reply:
That's neat! Updated.